nemo/collections/nlp/models/token_classification/ner_model.py

```diff
@@ -54,36 +54,59 @@ def __init__(self, cfg: DictConfig, trainer: Trainer = None):
             data_dir=cfg.data_dir, modes=["train", "test", "dev"], pad_label=cfg.pad_label
         )
         self.data_dir = cfg.data_dir
-        self.model_cfg = cfg
+        # self.model_cfg = cfg
```

Comment:
@titu1994  and @ekmb what do you think about this?

Reply:
I think this is good but slightly problematic? So the logic is:
 - If path exists, 
    - get its file basename
    - test again if file exists at ("./{basename}")? 

This is problematic cause we generally store vocabs somewhere else. So the second "if case" will fail unless we copy the vocab dir to the exact location where the code is run (which we can do, but more logic is needed).

Reply:
Can we do a double check - the first check if the file is resolved, great, set `vocab_file={cfg.language_model.vocab_file}`.

If that check fails, then compute base name and check for second test (as in how the serialization code will store directly inside the current execution dir so this time it will succeed).

If both fail, raise file not found error?

Reply:
After discussion, this looks like good logic.

Reply:
refactored into separate function after discussion

Reply:
@okuchaiev Sorry, missed the question. Why do you need tokenizer_model - it's only used for sentence piece. Most of the models work with Nemobert tokenizer and use only pretrained_model_name for init. I'd suggest initializing the tokenizer first and them pull the vocab out of it and store it in artifacts, maybe store vocab + tokenizer cls name, but not the model

Reply:
@ekmb it stores whatever artifacts were provided using during model instantiation. So if vocab were provided it should store it, if tokenizer_model were provided it will store that. If either is None it will not add it to .nemo file on saving